models/news.model.js

```diff
@@ -9,21 +9,36 @@ const readTopics = () => {
     });
 };
 
-const readArticles = () => {
+const readArticles = ((req, sortBy = "created_at", order = "DESC")=>{
+  const { topic } = req.query;
+
+  if (!topic){ 
   let countString =  `SELECT articles.article_id, articles.title, articles.topic, 
   articles.author, articles.body, articles.created_at, articles.votes, 
   articles.article_img_url,  COUNT(comments.body) AS comment_count
   FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id
-  GROUP BY articles.article_id ORDER BY articles.created_at DESC;` 
-  
-  return db.query(countString)
-    .then((result) => {
-    return result.rows;
-    }).catch((err) => {
-      console.log(err);
-    });
-};
-
+  GROUP BY articles.article_id ORDER BY ${sortBy}  ${order};` 
+  return db
+      .query(countString)
+      .then((result) => {
+        return result.rows;
+      })
+      
+  } else if (topic) {
+    return db
+      .query(
+        `SELECT * FROM articles WHERE topic = $1 ORDER BY ${sortBy} ${order}; `,
+        [topic]
+      )
+      .then((result) => {
+        if (result.rows.length === 0) {
+          return Promise.reject({ status: 404, msg: "Path not found" });
+        } else {
+          return result.rows;
+        }
+      });
```

Comment:
There should not be two queries here this can definetly be refactored to have a single query and instead building up the query string based on the user input given.